Gems/Atom/RPI/Code/Source/RPI.Public/RenderPipeline.cpp

```diff
@@ -366,6 +385,7 @@ namespace AZ
 
         void RenderPipeline::SetPersistentView(const PipelineViewTag& viewTag, ViewPtr view)
         {
+            AZ_Assert(view, "Persistent View for ViewTag [%s] is invalid.", viewTag.GetCStr());
```

Comment:
Is there an issue if we allow user to set the persistent view to empty (nullptr)?

From my discord comment:
"I think the consideration was that RenderPipeline doesn't own the Views which its viewtag associate to. So it's possible there wasn't a View or the View was destructed. We should still allow the existence of a RenderPipeline object which doesn't reference any View objects. And we need this function to remove the reference incase the View is about to be destructed."

Reply:
`RenderPipeline::OnStartFrame()` will crash, since it has this snippet that doesn't check if the view is a nullptr:

```
else if (pipelineViews.m_type == PipelineViewType::Persistent)
{
    pipelineViews.m_views[0]->SetPassesByDrawList(&pipelineViews.m_passesByDrawList);
}
```
Adding a check there would be trivial, but I'm not sure that's the only place that would crash.
This also suggests setting an persistent view to empty wasn't properly supported so far, which is why I added the assert.

I did add the function `UnregisterView()` a while back, but that doesn't help if the view was already destructed and the reference is already invalid.
One solution would be to allow the nullptr, and handle that it similar to that function, and reset the `PipelineViewType` to `Unknown`. Would that break anything else that I'm not aware of now?

Reply:
FInally got arround to reseting the `PipelineViewType` to `Unknown`.